### PR TITLE
Reject absolute path when saving external data

### LINKED ISSUE #5552: Absolute path should be rejected in `onnx.save_model(localtion=)`
# Bug Report

### Is the issue related to model conversion?
No

### Describe the bug

The Python API `onnx.save_model` should reject if the `location` argument is provided with an absolute path since the [proto](https://github.com/onnx/onnx/blob/main/onnx/onnx.proto#L600) says

```
  // - "location" (required) - POSIX filesystem path relative to the directory where the ONNX
  //                           protobuf model was stored
```

A model with external data of absolute path is likely to fail if been moved around.

### System information

Not system dependented.

### Reproduction instructions

Run the attached script to generate model file.
The path should be rejected.
*Note* you need to modify the path to your local path for reproducing.

```py
import numpy as np
import onnx
from onnx import helper
from onnx import numpy_helper
from onnx import AttributeProto, TensorProto, GraphProto

M = 8
N = 16
K = 64

A = helper.make_tensor_value_info('A', TensorProto.FLOAT, [M, K])
b_data = np.random.uniform(-1, 1, size=K*N).reshape((K, N)).astype('float32')
B = numpy_helper.from_array(b_data, name='B')
C = helper.make_tensor_value_info('C', TensorProto.FLOAT, [M, N])

op = helper.make_node(
    'MatMul',
    ['A', 'B'],
    ['C'],
)

g = helper.make_graph(
    [op],
    'test-model',
    [A],
    [C],
    initializer=[B],
)

m = helper.make_model(g, producer_name='onnx-example')
onnx.save_model(m, 'repro.onnx', save_as_external_data=True, location='/home/zhenhuaw/playground/dl/graph/matmul/repro.data', size_threshold=0)

```

### Expected behavior

* Raise error like "The external data path you provided is an absolute path, please use relative path.".
* No model generated.

### PR DESCRIPTION
**Pull request by zhenhuaw-me**
Respect the proto def. Close #5552.

### CODE REVIEW

File: onnx/external_data_helper.py [hunk 1]
```diff
@@ -128,6 +129,8 @@ def convert_model_to_external_data(
     if all_tensors_to_one_file:
         file_name = str(uuid.uuid1())
         if location:
+            if os.path.isabs(location):
```
Comment (xadupre) [inline onnx/external_data_helper.py:132]: Should we allow absolute path relative to the model path?

Comment (zhenhuaw-me) [reply onnx/external_data_helper.py:132]: I think not. Absolute path will leads to problems like "file not found" if we move models around, which should be common.

Comment (zhenhuaw-me) [reply onnx/external_data_helper.py:132]: And, path `A` can always be a relative to path `B` on a given system.

**Review by jcwchen — APPROVED** [review]
Thank you for the PR! Only allowing relative path makes sense to me. Might be worthy to mention this limitation in the document for external data as well: https://github.com/onnx/onnx/blob/main/docs/PythonAPIOverview.md#loading-an-onnx-model-with-external-data.

Another document actually has mentioned the location should be relative path already: https://github.com/onnx/onnx/blob/main/docs/ExternalData.md#external_data-field.

**Pull request merged**